logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/auth/LogoutUseCase.kt

```diff
@@ -33,7 +33,7 @@ class LogoutUseCaseImpl @Suppress("LongParameterList") constructor(
         deregisterTokenUseCase()
         logoutRepository.logout()
         sessionRepository.logout(userId = userId, reason)
-        if (reason == LogoutReason.SELF_HARD_LOGOUT) {
+        if (reason == LogoutReason.SELF_HARD_LOGOUT || reason == LogoutReason.REMOVED_CLIENT || reason == LogoutReason.DELETED_ACCOUNT) {
```

Comment:
@typfel @MohamadJaara correct me if I'm wrong but I think that we should also remove client and user data when the client has been removed or account has been deleted, not only when there is a hard logout 🤔

Reply:
user is only removed on hard logout
when REMOVED_CLIENT or DELETED_ACCOUNT only client data need to be removed 
on SESSION_EXPIRED or SOFT_LOGOUT no data is removed

Reply:
Ok, thanks, so I changed it to this:
```
        if (reason == LogoutReason.REMOVED_CLIENT || reason == LogoutReason.DELETED_ACCOUNT || reason == LogoutReason.SELF_HARD_LOGOUT) {
            // we put this delay here to avoid a race condition when receiving web socket events at the exact time of logging put
            delay(CLEAR_DATA_DELAY)
            clearClientDataUseCase()
        }
        if (reason == LogoutReason.SELF_HARD_LOGOUT) {
            clearUserDataUseCase() // this clears also current client id
        } else {
            clientRepository.clearCurrentClientId()
        }
```
It should be ok now, right?

Reply:
Ok, thanks @MohamadJaara , so I changed it to this:
```
        if (reason == LogoutReason.REMOVED_CLIENT || reason == LogoutReason.DELETED_ACCOUNT || reason == LogoutReason.SELF_HARD_LOGOUT) {
            // we put this delay here to avoid a race condition when receiving web socket events at the exact time of logging put
            delay(CLEAR_DATA_DELAY)
            clearClientDataUseCase()
        }
        if (reason == LogoutReason.SELF_HARD_LOGOUT) {
            clearUserDataUseCase() // this clears also current client id
        } else {
            clientRepository.clearCurrentClientId()
        }
```
It should be ok now, right?